src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -0,0 +1,298 @@
+                }
+
+                Debug.Fail("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Decrypt", "Argument 'offset' out of range.");
+
+                throw new ArgumentOutOfRangeException("offset");
+            }
+
+            if (count < 0 || count > (buffer == null ? 0 : buffer.Length - offset))
+            {
+                if (GlobalLog.IsEnabled)
+                {
+                    GlobalLog.Assert("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Decrypt", "Argument 'count' out of range.");
+                }
+
+                Debug.Fail("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Decrypt", "Argument 'count' out of range.");
+
+                throw new ArgumentOutOfRangeException("count");
+            }
+
+            newOffset = offset;
+            return Interop.GssApi.Decrypt(((SafeDeleteNegoContext)securityContext).GssContext, buffer, offset, count);
+        }
+
+        internal static int DecryptNtlm(
+            SafeDeleteContext securityContext,
+            byte[] buffer,
+            int offset,
+            int count,
+            bool isConfidential,
+            out int newOffset,
+            uint sequenceNumber)
+        {
+            throw new PlatformNotSupportedException(SR.net_nego_ntlm_not_supported);
+        }
+
+        private static SecurityStatusPal EstablishSecurityContext(
+          SafeFreeNegoCredentials credential,
+          ref SafeDeleteContext context,
+          bool isNtlm,
+          string targetName,
+          ContextFlagsPal inFlags,
+          SecurityBuffer inputBuffer,
+          SecurityBuffer outputBuffer,
+          ref ContextFlagsPal outFlags)
+        {
+            Debug.Assert(!isNtlm, "EstablishSecurityContext: NTLM is not yet supported");
+
+            if (context == null)
+            {
+                context = new SafeDeleteNegoContext(credential, targetName);
+            }
+
+            SafeDeleteNegoContext negoContext = (SafeDeleteNegoContext)context;
+            try
+            {
+                Interop.NetSecurityNative.GssFlags inputFlags = ContextFlagsAdapterPal.GetInteropFromContextFlagsPal(inFlags);
+                uint outputFlags;
+                SafeGssContextHandle contextHandle = negoContext.GssContext;
+                bool done = Interop.GssApi.EstablishSecurityContext(
+                   ref contextHandle,
+                   credential.GssCredential,
+                   isNtlm,
+                   negoContext.TargetName,
+                   inputFlags,
+                   ((inputBuffer != null) ? inputBuffer.token : null),
+                   out outputBuffer.token,
+                   out outputFlags);
+
+                Debug.Assert(outputBuffer.token != null, "Unexpected null buffer returned by GssApi");
+                outputBuffer.size = outputBuffer.token.Length;
+                outputBuffer.offset = 0;
```

Comment:
We just called an interop method that built outputBuffer; why do we need to overwrite values here?